cmd/cve/validate/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/goval_dictionary"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/nvd"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/oval"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+func main() {
+	dbDir := flag.String("db_dir", "/tmp/vulndbs", "Path to the vulnerability database")
+	debug := flag.Bool("debug", false, "Sets debug mode")
+	flag.Parse()
+
+	logger := log.NewJSONLogger(os.Stdout)
+	if *debug {
+		logger = level.NewFilter(logger, level.AllowDebug())
+	} else {
+		logger = level.NewFilter(logger, level.AllowInfo())
+	}
+
+	vulnPath := *dbDir
+	checkNVDVulnerabilities(vulnPath, logger)
+	checkGovalDictionaryVulnerabilities(vulnPath)
+}
+
+func checkNVDVulnerabilities(vulnPath string, logger log.Logger) {
+	metaMap := make(map[string]fleet.CVEMeta)
+	if err := nvd.CVEMetaFromNVDFeedFiles(metaMap, vulnPath, logger); err != nil {
```

Comment:
Can you include a patch in the underlying ticket that would repro both the bounds check failure that we saw back in August and the associated feed gen bug, as a way to test this? Seems like that'd speed QA along significantly.

Reply:
https://app.zenhub.com/workspaces/g-software-current-sprint-67685f6ff1830a000f347a73/issues/gh/fleetdm/fleet/21300#issuecomment-2715855922